samtranslator/model/eventsources/push.py

```diff
@@ -747,8 +747,9 @@ def _add_swagger_integration(self, api, function, intrinsics_resolver):
 
             if self.Auth.get("ResourcePolicy"):
                 resource_policy = self.Auth.get("ResourcePolicy")
+                api_id = self.RestApiId if isinstance(self.RestApiId, string_types) else self.RestApiId.get("Ref")
```

Comment:
Is this to support specifying `RestApiId` as a pure string? If so why should we over failing? This pattern is not in line with CloudFormation, so struggling why we think this is the right thing to do.

Reply:
It seems like `RestApiId` can already be either string or dict (correct me if I'm wrong), but this part of the code assumed it to be a dict. So I just updated it to handle strings. If you look in the test file below, in my other changes, the `setUp` already had a string api id. Also, `samtranslator/validator/sam_schema/schema.json` seems to indicate it can be string or object (https://github.com/aws/serverless-application-model/blob/5511198db567a3bae496c35fb44de8994597e5d7/samtranslator/validator/sam_schema/schema.json#L324)

Reply:
Thought it was odd but looks like we have tests for that already: https://github.com/aws/serverless-application-model/blob/develop/tests/translator/input/api_cache.yaml#L12

So I yield this point. :)

Reply:
So instead, is there a way we could encapsulate this logic so that callers getting the RestApiId get it in a standard way? If we have places we already handle this, adding more places just keeps that pattern going and risk when we add something new, it might be missed in other spots.